profile/src/v100/factors/is_factor_source.rs

```diff
@@ -0,0 +1,12 @@
+use crate::prelude::*;
+
+pub trait BaseIsFactorSource:
```

Comment:
The enum `FactorSource` impls this trait, however, it cannot impl `IsFactorSource` trait below, since it requires static `fn kind() -> FactorSourceKind`.